src/toxcore/dht/dht_node.rs

```diff
@@ -123,6 +123,20 @@ impl DhtNode {
             }
         }
     }
+
+    /// convert Dhtnode to PackedNode object based on is_ipv6_enabled flag
+    pub fn to_packed_node(self, is_ipv6_enabled: bool) -> Option<PackedNode> {
+        if let Some(saddr) = self.get_socket_addr(is_ipv6_enabled) {
+            Some(PackedNode {
+                pk: self.pk,
+                saddr,
+            })
+        } else {
+            warn!("to_packed_node() fails because get_socket_addr returns None");
```

Comment:
I think this warning can be moved to `get_socket_addr` since it should never return `None`. And if you do this then this `if` can be converted to simple `map`.

Reply:
done